pkg/operator/controller/ingress/status.go

```diff
@@ -392,69 +458,29 @@ func computeIngressDegradedCondition(conditions []operatorv1.OperatorCondition)
 		},
 	}
 
-	var graceConditions, degradedConditions []*operatorv1.OperatorCondition
-	now := clock.Now()
-	for _, expected := range expectedConditions {
-		condition, haveCondition := conditionsMap[expected.condition]
-		if !haveCondition {
-			continue
-		}
-		if condition.Status == expected.status {
-			continue
-		}
-		failedPredicates := false
-		for _, ifCond := range expected.ifConditionsTrue {
-			predicate, havePredicate := conditionsMap[ifCond]
-			if !havePredicate || predicate.Status != operatorv1.ConditionTrue {
-				failedPredicates = true
-				break
-			}
-		}
-		if failedPredicates {
-			continue
-		}
-		if expected.gracePeriod != 0 {
-			t1 := now.Add(-expected.gracePeriod)
-			t2 := condition.LastTransitionTime
-			if t2.After(t1) {
-				d := t2.Sub(t1)
-				if len(graceConditions) == 0 || d < requeueAfter {
-					// Recompute status conditions again
-					// after the grace period has elapsed.
-					requeueAfter = d
-				}
-				graceConditions = append(graceConditions, condition)
-				continue
-			}
-		}
-		degradedConditions = append(degradedConditions, condition)
+	// Cover the rare case of no conditions
+	if len(conditions) == 0 {
+		return operatorv1.OperatorCondition{Type: operatorv1.OperatorStatusTypeDegraded, Status: operatorv1.ConditionFalse}, nil
 	}
-
+	graceConditions, degradedConditions, requeueAfter := checkConditions(expected, conditions)
 	if len(degradedConditions) != 0 {
 		// Keep checking conditions every minute while degraded.
-		requeueAfter = time.Minute
-
-		var degraded string
-		for _, cond := range degradedConditions {
-			degraded = degraded + fmt.Sprintf(", %s=%s (%s: %s)", cond.Type, cond.Status, cond.Reason, cond.Message)
-		}
-		degraded = degraded[2:]
+		retryAfter := time.Minute
 
+		degraded := formatConditions(degradedConditions)
 		condition := operatorv1.OperatorCondition{
 			Type:    operatorv1.OperatorStatusTypeDegraded,
 			Status:  operatorv1.ConditionTrue,
 			Reason:  "DegradedConditions",
 			Message: "One or more other status conditions indicate a degraded state: " + degraded,
 		}
 
-		return condition, retryableerror.New(errors.New("IngressController is degraded: "+degraded), requeueAfter)
+		return condition, retryableerror.New(errors.New("IngressController is degraded: "+degraded), retryAfter)
 	}
-
 	condition := operatorv1.OperatorCondition{
 		Type:   operatorv1.OperatorStatusTypeDegraded,
 		Status: operatorv1.ConditionFalse,
 	}
-
 	var err error
 	if len(graceConditions) != 0 {
 		var grace string
```

Comment:
We could simplify this as `grace := formatConditions(graceConditions)`.

Reply:
The format is different for reporting graceConditions than reporting degradedConditions, so I did not use the method here.
grace: fmt.Sprintf(", %s=%s", cond.Type, cond.Status)
degraded: fmt.Sprintf(", %s=%s (%s: %s)", cond.Type, cond.Status, cond.Reason, cond.Message)